xcat-inventory/xcclient/inventory/manager.py

```diff
@@ -71,18 +72,45 @@ def getDBInst(self):
             InventoryFactory.__db__=dbfactory(self.dbsession)
         return InventoryFactory.__db__
 
-    def exportObjs(self, objlist, location=None):
+    def exportObjs(self, objlist, location=None,fmt='json'):
         myclass = InventoryFactory.__InventoryClass__[self.objtype]
         myclass.loadschema(self.schemapath)
         tabs=myclass.gettablist()
         obj_attr_dict = self.getDBInst().gettab(tabs, objlist)
         objdict={}
+        myobjdict2dump={}
         objdict[self.objtype]={}
         for key, attrs in obj_attr_dict.items():
             if not key:
                continue
             newobj = myclass.createfromdb(key, attrs)
-            objdict[self.objtype].update(newobj.getobjdict())
+            myobjdict=newobj.getobjdict()
+            myobjdict2dump[self.objtype]=myobjdict
+            objdict[self.objtype].update(myobjdict)
+            osimagefiles=newobj.getfilestosave()
+            if location: 
+                mydir=location+'/'+key
+                if os.path.exists(mydir):
+                    shutil.rmtree(mydir)
+                os.mkdir(mydir) 
+                myfile=mydir+'/'+'definition.yaml'
+                if fmt=='yaml':
+                    myfile=mydir+'/'+'definition.yaml'
+                elif fmt=='json':
+                    myfile=mydir+'/'+'definition.json'
+                dumpobj(myobjdict2dump,fmt,myfile) 
+                for imgfile in osimagefiles:
+                    if os.path.exists(imgfile):
+                        dstfile=mydir+imgfile
+                        try:
+                            os.makedirs(os.path.dirname(dstfile))
+                        except OSError, e:
+                            if e.errno != os.errno.EEXIST:
+                                raise
+                            pass
+                        shutil.copyfile(imgfile,dstfile)
+                    else:
+                        print("Warning: The file \""+imgfile+"\" of "+self.objtype+" object \""+key+"\" does not exist",file=sys.stderr) 
         return objdict
```

Comment:
Using more pythonic way here `"Warning: The file \"%s\" of %s object \"%s\" does not exist" % (imgfile, self.objtype, key)`